### PR TITLE
issues1261 fix md5sum in index.json

### PR DESCRIPTION
**Pull request by samba2**

### CODE REVIEW

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt [hunk 1]
```diff
@@ -29,67 +30,82 @@ import org.factcast.schema.registry.cli.utils.mapEventTransformations
...
 
         return Index(schemas, transformations.plus(syntheticTransformations))
     }
+
+    private fun createMd5Hash(filePath: Path, schemaStripTitles: Boolean): String =
+            if (schemaStripTitles) createTitleFilteredMd5Hash(filePath)
+                else checksumService.createMd5Hash(filePath)
+
+    private fun createTitleFilteredMd5Hash(filePath: Path): String {
+        val filteredJsonNode = titleFilterService.filter(
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt:106]: using a "service" here is probably overblown as it must not have state, and does not have dependencies either.
a simple function would do here.

Comment (uweschaefer) [reply factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt:106]: Also, TitleFilterServiceImpl deals with nullable, which it should not?

**Review by uweschaefer — APPROVED** [review]
lgtm

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt (hunk 1 above)
Comment (samba2) [reply factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt:106]: > using a "service" here is probably overblown as it must not have state, and does not have dependencies either.
> a simple function would do here.

I totally agree. For some reasons we've implemented it as a filter in the first place. You're right - sharing a stateless function can be done much easier.

Comment (uweschaefer) [reply factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt:106]: @samba2 do you want to commit the nullability issue?

**Pull request merged**